airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2698,16 +2698,17 @@ def execute(self, context: Any):
             self._handle_job_error(job)
 
             return self.job_id
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=BigQueryInsertJobTrigger(
-                conn_id=self.gcp_conn_id,
-                job_id=self.job_id,
-                project_id=self.project_id,
-                poll_interval=self.poll_interval,
-            ),
-            method_name="execute_complete",
-        )
+        elif self.deferrable and job.running():
```

Comment:
1.Corrected in d443fa3
2.If there's an error it will be sent to `execute_complete()` from `BigQueryInsertJobTrigger`.The task will fail due to the AirflowException raised there.